### PR TITLE
Add poetry to manage dependencies

### PR DESCRIPTION
**Pull request by naddeoa**
This change introduces poetry as a dependency management tool. Poetry will be replacing our requirements files, allow us to specify dev specific dependencies in a managed way, and generate lock files so we can reproduce installations on other machines.

### CODE REVIEW

**Comment by andyndang** [comment]
```
Running setup.py develop for whylogs
    ERROR: Command errored out with exit status 1:
     command: /Users/runner/work/whylogs/whylogs/.tox/py39/bin/python -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/Users/runner/work/whylogs/whylogs/setup.py'"'"'; __file__='"'"'/Users/runner/work/whylogs/whylogs/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' develop --no-deps
         cwd: /Users/runner/work/whylogs/whylogs/
    Complete output (3 lines):
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
    ModuleNotFoundError: No module named 'setuptools'
    ----------------------------------------
ERROR: Command errored out with exit status 1: /Users/runner/work/whylogs/whylogs/.tox/py39/bin/python -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/Users/runner/work/whylogs/whylogs/setup.py'"'"'; __file__='"'"'/Users/runner/work/whylogs/whylogs/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' develop --no-deps Check the logs for full command output.

```
CI is failing atm

File: tests/unit/io/test_file_loader.py [hunk 1]
```diff
@@ -19,9 +19,11 @@ def test_image_loader(image_files):
 
 @pytest.mark.skipif(sys.version_info < (3, 8), reason="requires python3.7 or higher")
 def test_file_loader(file_list):
+    filefmts = [(pd.DataFrame, "csv"), (list, "jsonl")]
```
Comment (lalmei) [inline tests/unit/io/test_file_loader.py:22]: I dont see the need to support these old versions of python or pandas

Comment (lalmei) [reply tests/unit/io/test_file_loader.py:22]: especially when it removes functionality

Comment (naddeoa) [reply tests/unit/io/test_file_loader.py:22]: I agree, my current plan is to get it working and checked in with the version that we have already (this bad one) and bring it up again to talk about dropping support for 3.6 for supporting excel in a different way than pandas.

**Review by andyndang — APPROVED** [review]
LGTM

**Pull request merged**